analyzer/tests/libtest/decorators.py

```diff
@@ -0,0 +1,40 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+"""
+Test decorators.
+"""
+
+from functools import wraps, partial
+
+
+NO_CTU_MESSAGE = "CTU is not supported"
+NO_CTU_ON_DEMAND_MESSAGE = "CTU-on-demand is not supported"
+NO_CTU_DISPLAY_PROGRESS_MESSAGE = "CTU diplay progress is not supported"
+
+
+def makeSkipUnlessAttributeFound(attribute, message):
+    def decorator(original):
+        @wraps(original)
+        def wrapper(self, *args, **kwargs):
+            if not getattr(self, attribute):
+                self.skipTest(message)
+            else:
+                original(self, *args, **kwargs)
+        return wrapper
+    return decorator
+
+
+makeSkipUnlessCTUCapable = partial(makeSkipUnlessAttributeFound,
```

Comment:
These decorators are very CTU specific, what do you think about renaming the module to `ctu_decorators`?

For the general `makeSkipUnlessAttributeFound` a separate decorators could be created if it is used in other tests but I see it only in the ctu related tests now. @csordasmarton what do you think?

Reply:
I have introduced generic decorators and a ctu_decorators module, maybe this improves reusability :)